python/cuml/test/test_nearest_neighbors.py

```diff
@@ -507,6 +507,31 @@ def test_knn_graph(input_type, mode, output_type, as_instance,
         assert isspmatrix_csr(sparse_cu)
 
 
+@pytest.mark.parametrize('nrows', [unit_param(500), stress_param(70000)])
+def test_nearest_neighbors_rbc(nrows):
+    X, y = make_blobs(n_samples=nrows,
+                      n_features=2, random_state=0)
+
+    knn_cu = cuKNN(metric="euclidean", algorithm="rbc")
+    knn_cu.fit(X)
+
+    rbc_d, rbc_i = knn_cu.kneighbors(X[:int(nrows/2), :])
+
+    knn_cu2 = cuKNN(metric="sqeuclidean", algorithm="brute")
+    knn_cu2.fit(X)
+
+    brute_d, brute_i = knn_cu2.kneighbors(X[:int(nrows/2), :])
+
+    cp.testing.assert_allclose(rbc_d, brute_d, atol=5e-2,
+                               rtol=1e-1)
+    rbc_i = cp.sort(rbc_i, axis=1)
+    brute_i = cp.sort(brute_i, axis=1)
+
+    diff = rbc_i != brute_i
+
+    assert diff.ravel().sum() < 5
```

Comment:
Curious how did you choose 5?

Reply:
Mostly empirically. It's not so much important that these values match exactly so long as the distances match exactly (to a reasonable precision). Idea here is that the actual neighbors can be nondeterministic in the face of distances which are the same or very close.